.github/workflows/build.yaml

```diff
@@ -0,0 +1,53 @@
+name: Build and verify
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  push:
+    branches: [$default-branch]
+permissions:
+  contents: read
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        include:
+          - jdk: 11.0.16
+            sonarEnabled: true
+          - jdk: 17.0.4
+            sonarEnabled: false
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3.0.2
+        with:
+          # Fetch entire history for SonarCloud analysis.
+          fetch-depth: 0
+      - name: Set up JDK
+        uses: actions/setup-java@v3.5.1
+        with:
+          java-version: ${{ matrix.jdk }}
+          distribution: temurin
+          cache: maven
+      - name: Cache SonarCloud packages
+        if: ${{ matrix.sonarEnabled }}
+        uses: actions/cache@v3.0.8
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+      - name: Display build environment details
+        run: mvn --version
+      - name: Build the project
+        run: mvn -T1C install
+      - name: Perform SonarCloud analysis
+        if: ${{ matrix.sonarEnabled }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: >-
+          mvn sonar:sonar
+          -Dsonar.host.url=https://sonarcloud.io
+          -Dsonar.organization=picnic-technologies
+          -Dsonar.projectKey=tech.picnic:oss-parent
```

Comment:
The first two will also apply downstream; we can add them to `pom.xml`. The third _should_ be inferred; let me test that :eyes:

Reply:
Seems to work :smile: